### DISCUSSION TITLE
Autocomplete: accept array of strings

### DISCUSSION THREAD

**Discussion by cstrat** [mantinedev/mantine#171] [Feedback]
This might also be applicable for `Select`, but makes perfect sense for Autocomplete.

Would it be possible to allow the `data` prop to accept an array of strings or the array of objects?
Having an array of objects which are simply: `[{ value: "xxx" }]` could be short-circuited with just `['xxx']`

I've already run into this a couple times where I've forgotten I need to use the object. The component doesn't handle the incorrectly formatted prop very well.

https://codesandbox.io/s/mantine-autocomplete-fbete

I'm happy to try a PR with a helper function which converts the prop from an array of strings to the formatted object array - I just don't know if that has an impact to the typescript config.

**Comment by rtivital** [comment]
Yep, sure that would be a great addition and not that hard to implement, you'll need to add some more logic for TypeScript:

- change type:
```tsx
type AutocompleteItem = string | {
  value: string;
  [key: string]: any;
}
```
- and add logic to transform strings to an array of objects

**Comment by rtivital** [comment]
As I continue thinking about it – this can be also done for select so if you will work on that please also add this feature

**Reply by cstrat** [reply]
Yeah I thought that too. Map the strings to be both the `label` and `value`.
